components/style/values/specified/font.rs

```diff
@@ -18,6 +18,90 @@ use values::specified::length::{AU_PER_PT, AU_PER_PX, FontBaseSize};
 
 const DEFAULT_SCRIPT_MIN_SIZE_PT: u32 = 8;
 
+#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, PartialEq, ToCss)]
+/// A specified font-weight value
+pub enum FontWeight {
+    /// Normal variant
+    Normal,
+    /// Bold variant
+    Bold,
+    /// Bolder variant
+    Bolder,
+    /// Lighter variant
+    Lighter,
+    /// Computed weight variant
+    Weight(computed::FontWeight),
+    /// System font varaint
+    System(SystemFont),
+}
+
+impl FontWeight {
+    /// Get a specified FontWeight from a gecko keyword
+    pub fn from_gecko_keyword(kw: u32) -> Self {
+        computed::FontWeight::from_int(kw as i32).map(FontWeight::Weight)
+            .expect("Found unexpected value in style struct for font-weight property")
+    }
+
+    /// Get a specified FontWeight from a SystemFont
+    pub fn system_font(f: SystemFont) -> Self {
+        FontWeight::System(f)
+    }
+
+    /// Retreive a SystemFont from FontWeight
+    pub fn get_system(&self) -> Option<SystemFont> {
+        if let FontWeight::System(s) = *self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+}
+
+impl Parse for FontWeight {
+    fn parse<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>)
+                     -> Result<FontWeight, ParseError<'i>> {
+        let result = input.try(|input| {
```

Comment:
This could just be something like:

```rust
Ok(match *input.next()? {
    Token::Ident(ref ident) => {
        match_ignore_ascii_case! { ident,
            "normal" => FontWeight::Normal,
            // ...
            _ => return Err(input.new_custom_error(..)),
        }
    }
    Token::Number { int_value: Some(value), .. } => {
        computed::FontWeight::from_int(value)?;
    }
})
```

Which is clearer I think, and also more efficient.

Also, could you note that the grammar of this property intentionally doesn't support `calc()`?